default/python/AI/ProductionAI.py

```diff
@@ -2033,3 +1697,285 @@ def _build_military_command() -> float:
         # default selection should prefer the capital, unless its species dislikes it.
         return _try_enqueue(building_type, get_inhabited_planets())
     return 0.0
+
+
+def _build_basic_shipyards() -> Tuple[
```

Comment:
I'd say let's make NamedTuple here. Return structure is quite complicated, and a named tuple will document it well.